src/dataset_dog/api.py

```diff
@@ -39,7 +38,22 @@ def _submit_callback(
         )
         self.worker.submit(function_info)
 
-    def record_function(self, frequency: float):
+    @staticmethod
+    def filter_arguments(args: Tuple[str], kwargs: Dict[str, Any], skip_args: List[str]) -> Tuple[List[str], Dict[str, Any]]:
```

Comment:
args aren't a string tuple. They can be anything. You'll have to use the inspect module(https://stackoverflow.com/questions/218616/how-to-get-method-parameter-names) to get the args name and then do the filtering